src/type_conversions.rs

```diff
@@ -324,9 +329,9 @@ impl<'a> TryFrom<CompatiblePyType<'a>> for Any {
             CompatiblePyType::List(l) => {
                 let result: PyResult<Vec<Any>> = l
                     .into_iter()
-                    .map(|py_any|CompatiblePyType::try_from(py_any)?.try_into())
+                    .map(|py_any| CompatiblePyType::try_from(py_any)?.try_into())
                     .collect();
-                result.map(|res| Any::Array(res.into_boxed_slice()))
+                result.map(|res| Any::Array(Arc::from(res.into_boxed_slice())))
```

Comment:
I'm not sure if `.into_boxed_slice` is needed. Possibly `Arc::from(res)` or `Arc::from(res.as_slice())` could work as well without intermediate allocations.